lib/gpw/companies_details_updator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CompaniesDetailsUpdator < ApplicationService
+  def initialize
+    @gpw_client = GPW::Client.new
+  end
+
+  def call
+    update_details_of_gpw_companies
+    update_details_of_nc_companies
+  end
+
+  private
+
+  def update_details_of_gpw_companies
+    companies_attrs = @gpw_client.fetch_gpw_companies_details
+
+    update_companies_details(companies_attrs)
+  end
+
+  def update_details_of_nc_companies
+    companies = @gpw_client.fetch_nc_companies_details
+
+    update_companies_details(companies)
+  end
```

Comment:
If the `update_companies_details` is the same for all the companies, you could do:

```suggestion
  def call
    companies = fetch_companies_details
    update_companies_details(companies_attrs)
  end

  private

  def fetch_companies_details
    @gpw_client.fetch_gpw_companies_details + @gpw_client.fetch_nc_companies_details
  end
```